### PR TITLE
Markdown lint

### PR DESCRIPTION
**Pull request by peterbe**
~THIS NEEDS TO WAIT FOR https://github.com/mdn/content/pull/486~

~Once https://github.com/mdn/content/pull/486#event-4132883063 is merged, we can merge in main into this branch. Because that's what I started on.~

This workflow only wakes up if any of the `*.md` files are in the diff. 

This'll fail a lot until we tackle the extreme line lengths, which is a bit contentious because not everyone agrees that the line length should be max 80. One battle at a time.

### CODE REVIEW

**Comment by peterbe** [comment]
@nschonni lemme know when you get your PR merged and I'll update this PR here.

**Comment by nschonni** [comment]
I rebased the other one, but feel free to rebase and supersede next time

**Comment by vkWeb** [comment]
@peterbe #486 is merged. Now we are ready to get main merged in here and proceed.

**Comment by peterbe** [comment]
At the time of writing, this PR is actually passing. That's because the GHA workflow doesn't even run since my change doesn't include any changes to any `*.md` files. 

@chrisdavidmills @escattone Since you wrote large parts of the `README.md` and `REVIEWING.md` do you support moving to a capped line width? 

Personally, I think it's a good idea because when you look at PRs in GitHub it's difficult to scroll vertically. And when you break up large lines into multiple lines, it's easier to edit it from multiple different patches without causing conflicts.

File: .github/workflows/markdown-lint.yml [hunk 1]
```diff
@@ -0,0 +1,25 @@
+name: Markdown lint (project files)
+
+on:
+  pull_request:
+    branches:
+      - main
+    paths:
+      - '*.md'
```
Comment (nschonni) [inline .github/workflows/markdown-lint.yml:8]: In order to trigger on changes to the CI, I usually include this action path in here

Comment (peterbe) [reply .github/workflows/markdown-lint.yml:8]: Am I having deja vu or is that the same thing that @bershanskiy brought up when he and I worked on something similar. 

In fact, it was @bershanskiy who showed me about the `on.paths` feature.

Comment (peterbe) [reply .github/workflows/markdown-lint.yml:8]: You're right. I prefer this. 

@bershanskiy That reminds me, if you touch the `pr-build.yml` file, it triggers a build of that workflow but the `$GIT_DIFF` env var becomes empty, so it runs the whole `yarn build` without any filtering. Takes about an hour :)

**Comment by escattone** [comment]
@peterbe 👍 I do like the idea of capping the line length.

**Comment by Ryuno-Ki** [comment]
Is this still relevant? (Sorry, not meaning to sound harsh)

**Comment by peterbe** [comment]
> Is this still relevant? (Sorry, not meaning to sound harsh)

It is. But I first have to rewrite all the existing `.md` files so they no longer have crazy long lines.

**Comment by peterbe** [comment]
@chrisdavidmills r? 
Now all lines are within 80 chars and I corrected some bullet points to use `-` instead of `*`.

**Comment by peterbe** [comment]
Yes. Yes, it was painful to manually correct all of these lines. :)

**Review by chrisdavidmills — APPROVED** [review]
@peterbe ack, painful work to have to do; feel free to ask me for help in future with such things. 

But yeah, looks good, thanks @peterbe !

**Review by schalkneethling — APPROVED** [review]

**Comment by hamishwillee** [comment]
Just FMI, is it intended that the 80char limit would also apply to content markdown files (when we have them). I much prefer my markdown source to break on full stops :-)

**Comment by peterbe** [comment]
@hamishwillee Mayhaps. But it's potentially a difficult policy to get consensus on. When I type Markdown in my GitHub comments (like, right here right now) I don't care about having to insert newlines when it gets too long. 
But one also doesn't have to view diffs of comments texts.

**Comment by peterbe** [comment]
We'll see what's most convenient and least friction for the team.

**Comment by hamishwillee** [comment]
@peterbe My preference is somewhat based on past experience with translation software that treated separate lines as different translations strings (this pissed off translators). That is probably software-specific, and in any case I could equally argue for shorter strings because that makes editing in the github editor much easier. As long as the linter doesn't break just before a full stop or something similarly irritating I will survive. Cheers.

**Comment by peterbe** [comment]
I think it's an interesting enough topic. My opinions *will* depend on what we discover when we start migrating towards something Markdownish. 
For `.md` files you check into git repos and almost treat them like code, I think the 80 char rule is great because it makes the git diffs sensible and it's generally easier to reason around shorter lines as you display them in editors and stuff. 
And it's so annyoying here in mdn/content when you're reviewing a PR and some lines force you to scroll so far to the right that if you accidentally place the cursor else where it scrolls you all the way back to the left. 
And if a whole paragraph is one long monster-long line, won't you get git merge conflicts if two people are editing the same line but in two entirely different areas of the paragraph. 

But also, the paragraph I just wrote above actually just 4 lines, but displayed here in GitHub's UI, it displays as 7 lines. It was pleasant to be able to just type that without having to worry about where 80-chars is. And if I decide to go back and edit a couple of words, it could have forced me to rearrange an annoying "trickle-down effect" to get the following lines to also match 80 chars. For source code text, that's rarely a problem.

So we'll see. Unlike pure source code, it feels like the pros ~= cons for this rule for Markdown. We do expect some people to edit the future Markdown within their VSCode/Atom/JetBrains and those IDEs have nice markers to guide you. But a lot of people will just press the little "pen icon" to edit it right here in the GitHub UI and then you don't have that help. 

Perhaps a hybrid. Instead of 80, we pick 100. Or, perhaps of making it a hard rule, we figure out a way to display a warning to the PR author, in CI or in the Yari previwer, that "Your lines are a wee bit too long, but we'll let it fly this time. M'kay."

**Comment by Ryuno-Ki** [comment]
> And it's so annyoying here in mdn/content when you're reviewing a PR and some lines force you to scroll so far to the right that if you accidentally place the cursor else where it scrolls you all the way back to the left.

I actually have a custom user stylesheet (via Stylus) that I apply to enforce a softwrap on GitHub …

**Comment by Ryuno-Ki** [comment]
> Perhaps a hybrid. Instead of 80, we pick 100.

I _think_ Django has a soft limit of 80 chars and a hard one at 120 … (or was it Python in general?).

I agree with what you say. But I haven't considered the translation experience.

My own experience in that regard (using Crowdin or Transifex) is, that the whole file is translated.
If you were to use POEdit and GNU gettext, the lines to translate are wrapped by the development team.

**Pull request merged**